persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java

```diff
@@ -281,6 +281,11 @@ public boolean isInSchemaAgreement() {
         .build();
   }
 
+  @Override
+  public void executeAuthResponse(Runnable handler) {
+    executor.execute(handler);
```

Comment:
It looks like you are using this executor:
```
   executor =
        SHARED.newExecutor(
            DatabaseDescriptor.getNativeTransportMaxThreads(),
            Integer.MAX_VALUE,
            "transport",
            "Native-Transport-Requests");
```
responsible for transport. Is it a good one?
Shouldn't we consider creating a separate executor service responsible for processing auth requests and use It here?

Reply:
This is the same thread pool used by C* implementations.